lib/policyEvaluator/evaluator.js

```diff
@@ -133,11 +133,17 @@ function meetConditions(requestContext, statementCondition, log) {
             // reference_policies_multi-value-conditions.html)
             const keyBasedOnRequestContext =
                 findConditionKey(key, requestContext);
-            // Handle IfExists
+            // Handle IfExists and negation operators
+            const operatorsWithNegation = ['StringNotEquals',
+            'StringNotEqualsIgnoreCase', 'StringNotLike', 'ArnNotEquals',
+            'ArnNotLike', 'NumericNotEquals'];
+            const isNegationOperator =
+                operatorsWithNegation.indexOf(bareOperator) > -1;
             if ((keyBasedOnRequestContext === undefined ||
                 keyBasedOnRequestContext === null) &&
-                hasIfExistsCondition) {
-                log.trace('satisfies condition due to IfExists operator');
+                (hasIfExistsCondition || isNegationOperator)) {
+                log.trace('satisfies condition due to IfExists operator or ' +
+                'negation operator');
```

Comment:
one last comment, can you add the method attribute to the log? In this case, method should be
`evaluators.evaluatePolicy`